syntax = "proto3";
import "google/protobuf/wrappers.proto";
import "protos/base.proto";

package protos.connectors;

message PeriodicRunStatus {
  enum StatusType {
    UNKNOWN = 0;
    STARTED = 1;
    FINISHED = 2;
    ERROR = 3;
  }
}

message Connector {
  google.protobuf.UInt64Value id = 1;
  Source type = 2;
  google.protobuf.BoolValue is_active = 3;
  google.protobuf.StringValue name = 4;
  google.protobuf.StringValue created_by = 5;
  sfixed64 created_at = 6;
  sfixed64 updated_at = 7;
  google.protobuf.StringValue display_name = 8;
  google.protobuf.StringValue category = 9;
  repeated ConnectorKey keys = 10;
}

message ConnectorKey {
  google.protobuf.UInt64Value id = 1;
  SourceKeyType key_type = 2;
  google.protobuf.StringValue key = 3;
  google.protobuf.BoolValue is_active = 4;
  google.protobuf.UInt64Value connector_id = 5;
  sfixed64 created_at = 6;
  sfixed64 updated_at = 7;
  google.protobuf.StringValue display_name = 8;
}

message UpdateConnectorOp {
  enum Op {
    UNKNOWN = 0;
    UPDATE_CONNECTOR_NAME = 1;
    UPDATE_CONNECTOR_STATUS = 2;
    UPDATE_CONNECTOR_KEYS = 3;
  }

  message UpdateConnectorName {
    google.protobuf.StringValue name = 1;
  }

  message UpdateConnectorStatus {
    google.protobuf.BoolValue is_active = 1;
  }

  message UpdateConnectorKeys {
    repeated ConnectorKey connector_keys = 1;
  }

  Op op = 1;
  oneof update {
    UpdateConnectorName update_connector_name = 2;
    UpdateConnectorStatus update_connector_status = 3;
    UpdateConnectorKeys update_connector_keys = 4;
  }
}

message AccountActiveConnectorModelTypes {
  message ConnectorMetadataModelTypeMap {
    SourceModelType model_type = 1;
    google.protobuf.StringValue display_name = 2;
  }
  Source connector_type = 1;
  repeated ConnectorMetadataModelTypeMap model_types_map = 2;
}
